Source/codec.cpp

```diff
@@ -66,58 +66,37 @@ int __fastcall codec_decode(void *pbSrcDst, int size, char *pszPassword)
 
 void __fastcall codec_init_key(int unused, char *pszPassword)
 {
-	char *v2; // edi
-	char *v3; // esi
-	int v4; // eax
-	signed int v5; // ecx
-	char v6; // dl
-	unsigned int v7; // ecx
-	signed int v8; // esi
-	char v9[136]; // [esp+Ch] [ebp-E0h]
-	char v10[64]; // [esp+94h] [ebp-58h]
-	char dst[20]; // [esp+D4h] [ebp-18h]
-	int v12; // [esp+E8h] [ebp-4h]
+    int i;
+    char key[136]; // last 64 bytes are the SHA1
+    char pw[64];
+    char digest[SHA1HashSize];
 
-	v2 = pszPassword;
-	srand(0x7058u);
-	v3 = v9;
-	v12 = 136;
-	do
-	{
-		*v3++ = rand();
-		--v12;
-	}
-	while ( v12 );
-	v4 = 0;
-	v5 = 0;
-	do
-	{
-		if ( !v2[v4] )
-			v4 = 0;
-		v6 = v2[v4++];
-		v10[v5++] = v6;
-	}
-	while ( v5 < 64 );
-	SHA1Reset(0);
-	SHA1Calculate(0, v10, dst);
-	SHA1Clear();
-	v7 = 0;
-	do
-	{
-		v9[v7] ^= dst[(signed int)v7 % 20];
-		++v7;
-	}
-	while ( v7 < 0x88 );
-	memset(v10, 0, sizeof(v10));
-	memset(dst, 0, sizeof(dst));
-	v8 = 0;
-	do
-	{
-		SHA1Reset(v8);
-		SHA1Calculate(v8++, &v9[72], 0);
-	}
-	while ( v8 < 3 );
-	memset(v9, 0, sizeof(v9));
+    srand(0x7058);
+
+    char *keyInit = key;
+    for (i = 0; i < 136; i++) {
+        *keyInit = rand();
+        keyInit++;

```

Comment:
When using `key[i]` in the loop instead of `keyInit` pointer, do the bin-exactness suffer?

Reply:
Yeah i tried several different loops but didnt get to match as an array